### ISSUE TITLE
[SUGGESTION] effect for wait until condition is met

### ISSUE THREAD

**Issue by Fusezion** [SkriptLang/Skript#5768]
### Suggestion

Add a pattern for the `Delay` effect that allows a user to do `wait until <.+>`

### Why?

A user wouldn't shut up about it in the discord, as such it's basically a 1 liner while section that doesn't supporting writing any code inside it.

### Other

Unsure if this should be added to `SecWhile` as an effect section or as suggested into `Delay` class 

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this suggestion.

**Comment by TheLimeGlass** [comment]
Duplicate https://github.com/SkriptLang/Skript/issues/4000

**Comment by Moderocky** [comment]
This isn't actually a duplicate, #4000 describes an enhanced switch statement, not a wait-for behaviour.

(Admittedly, the word 'when' in 4000 implies the wrong thing.)

**Comment by TheLimeGlass** [comment]
> This isn't actually a duplicate, #4000 describes an enhanced switch statement, not a wait-for behaviour.
> 
> (Admittedly, the word 'when' in 4000 implies the wrong thing.)

It describes both

**Comment by Fusezion** [comment]
> It describes both

This is disagree with, the `when` switch statement they're wanting feels less like a section or effect and more like a structure as using it is weird inside a event
```
on block break:
    set {blank} to someFunctionWithAWait()
    when {blank}:
        is an integer -> broadcast "It's an integer"
    broadcast "Have we reached the end?"
```

I understand where you're coming from saying they're the same as they are generally very very similar but this feels better
```
on block break:
    set {blank} to someFunctionWithAWait()
    wait until {blank} is set
    broadcast "Okays o we've reached the end"
```

**Comment by Moderocky** [comment]
> It describes both

No, Mwexim's suggestion was for Kotlin's `when` block, which is a fancier version of Java's `switch` statement. It does **not** wait for something to be true.
Mwexim's suggestion is for a syntax equivalent to:
```
if {var} is 5:
    broadcast "yes"
if {var} is 10:
    broadcast "no"
```

Fuzesizion is asking for a wait-until-X effect (what you thought the #4000 issue was about).

**Comment by DelayedGaming** [comment]
Can't you just do:
```py
while {_p} is online:
    wait 5 ticks
```
If this suggestion is implemented, does the check interval is explicitly specified by Skript itself (1 tick at this point for accuracy) or it is configurable by users?

**Comment by Fusezion** [comment]
What you sent is basically exactly what it was suggested as, a `wait until <.+>` basically a wait effect with a condition that until met will not continue. for syntax it could probably be done like this

```applescript
wait until <.+> [(with|using) interval %-timespan%]
```
in the example you said it gave me the idea of doing what `target block` and `target entity` work where it has a config option
so internal without the interval setting will be whatever config option might be most likely `1 tick` default
with optional interval for a player to define if say they have a `wait until player is on ground` they can do `wait until ... with interval 5 ticks` 

In addition Ayham mentioned on discord adding something that will remove interval check and basically send true when an action happened I don't remember where this is so you might need to find it
